dockerfiles/semgrep-dev.Dockerfile

```diff
@@ -17,10 +17,13 @@ FROM returntocorp/semgrep:develop
 # Various utilities. We can always install them during the CI job but it's
 # it's nice to do it here while we're root.
 #
+# TODO: remove the explicit python3 install once our version of alpine has python3.11
+#
 RUN apk add --no-cache \
   bash \
   curl \
   jq && \
+  apk add --no-cache --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main python3 && \
   apk add --no-cache --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing pre-commit
```

Comment:
I have questions:
* why are these commands run in the same `RUN` instruction? If you don't know better than I do, let's break it into 3 RUNs.
* what requires python 3.11? -> comment
* adding a test for python 3.11:
```dockerfile
# install python 3.11 needed for XXX
RUN apk add ... && [[ "$(python3 --version)" =~ 3\.11\.[0-9]+ ]]
```
* while we're at it, let's add: `# TODO: explain why we need a specific version of pre-commit and what it version is`

Reply:
1. Chained `RUN` is a Dockerfile best practice. It reduces the number of layers and image size, which results in faster pulls. Let's bring this back in a future PR.
2. Thanks for adding the clarifying comment re: python 3.11 and pre-commit. I wonder if we even need pre-commit in this image? (Probably, I haven't taken the time to find where it's being used)
3. Two things:
- The `python3 --version` check isn't doing what you expect. The docker image (`alpine-python`) manually installs python (and other python-related things) outside of apk, so your check is not actually confirming that a satisfactory version of the `python3` apk package is installed. You'd need to check the output of something like `apk list --installed python3`.
- However, I'm not convinced that this explicit check is useful as-is, since it simply fails the build without relaying helpful information to the user. I feel that the original `apk` error output did a decent job telegraphing the problem (that `pre-commit` wants 3.11 but alpine only has a 3.10 package) and we were simply confused because the image provides an out-of-band version of 3.11. This will hopefully all become moot when alpine 3.18 is release and/or 3.11 is promoted to the `main` repository.